internal/gengapic/genrest.go

```diff
@@ -112,7 +159,59 @@ func (g *generator) genRESTMethod(servName string, serv *descriptor.ServiceDescr
 	}
 }
 
+func (g *generator) emptyUnaryRESTCall(servName string, m *descriptor.MethodDescriptorProto) error {
+	info, err := getHTTPInfo(m)
+	if err != nil || info == nil {
+		return err
+	}
+
+	inType := g.descInfo.Type[*m.InputType]
+	inSpec, err := g.descInfo.ImportSpec(inType)
+	if err != nil {
+		return err
+	}
+
+	p := g.printf
+	lowcaseServName := lowcaseRestClientName(servName)
+	p("func (c *%s) %s(ctx context.Context, req *%s.%s, opts ...gax.CallOption) error {",
+		lowcaseServName, m.GetName(), inSpec.Name, inType.GetName())
+
+	// TODO(dovs): handle cancellation, metadata, osv.
+	// TODO(dovs): handle http headers
+	// TODO(dovs): handle deadlines?
+	// TODO(dovs): handle call options
+	p("// The default (false) for the other options are fine.")
+	p("m := protojson.MarshalOptions{AllowPartial: true, Multiline: true, EmitUnpopulated: true}")
+	p("if jsonReq, err := m.Marshal(req); err != nil {")
+	p("  return err")
+	p("}")
+	p("")
+	// TODO(dovs): handle path parameters
+	p("url := fmt.Sprintf(\"https://%%s%s\", c.host)", info.url)
+	p("httpReq, err := http.NewRequestWithContext(ctx, \"%s\", url, bytes.NewReader(jsonReq))", strings.ToUpper(info.verb))
+	p("if err != nil {")
+	p("    return err")
+	p("}")
+	p("")
+	p("httpRsp, err := client.Do(httpReq)")
+	p("if err != nil{")
+	p(" return err")
+	p("} else if httpRsp.StatusCode >= 400 {")
+	p("  return errors.E(nil, httpRsp.Status)")
```

Comment:
`errors.E` isn't a standard golang API, its an internal one for the generator. There is https://golang.org/pkg/errors though.